src/odemis/acq/stream/_projection.py

```diff
@@ -821,9 +815,11 @@ def projectAsVis(self):
             except KeyError:
                 # Note: If this method called, while cache empty e.g. the bg image was loaded, which empties the cache.
                 # Recalculate raw should then always work.
-                if not self._polarimetry_cache_raw:
-                    _ = self.projectAsRaw()
-                data = self._project2RGBPolar(ebeam_pos, pol_pos, self._polarimetry_cache_raw[ebeam_pos])
+                try:
+                    raw = self._polarimetry_cache_raw[ebeam_pos]
+                except KeyError:
+                    raw = self._projectAsRaw(ebeam_pos)
```

Comment:
Sorry, I suggested a too complicated code. `_projectAsRaw()` already takes care of the cache. So you can always call it. => drop the try/except, and only keep the expect block.